clients/roscpp/src/libros/io.cpp

```diff
@@ -44,9 +44,16 @@
 #else
   #include <cstring> // strerror
   #include <fcntl.h> // for non-blocking configuration
+  #include <sys/epoll.h>
```

Comment:
Does that also mean we need to support all three implementation? Windows / poll / epoll ?
If that's the case, I'd much rather have an implementation file per OS. 
`io_win.cpp` / `io_posix.cpp` / `io_epoll.cpp`. 
That way, implementation file can be neat and clean (could be all in one file as well though.)